Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -1204,10 +1218,33 @@ extension MacroApplication {
 
 // MARK: Freestanding macro expansion
 
+protocol MacroExpansion<ResultType> {
+  associatedtype ResultType
+  func map<T>(_ transform: (ResultType) throws -> T) rethrows -> T
+}
+
```

Comment:
Is this protocol needed. I would have expected that it isn’t because the case in `MacroExpansionResult` could just be the `StackMacroExpansion` with the generic type specialized to `ResultType`.

Reply:
The reason why I used a protocol here instead of a concrete type is I want to make sure the only blessed way to access the expanded node is through the closure.

Reply:
You could just make everything but `map` private, right?

Reply:
> You could just make everything but `map` private, right?

Despite this, call sites in this file could still access the private fields?

Maybe I've gone too far to entertain my paranoia for discouraging unintended usages...

Reply:
No, `private` can only be accessed from the type itself, `fileprivate` can be accessed from anywhere in the file.

Reply:
I need to brush up my knowledge of Swift access modifiers...